crd-generator/src/main/java/io/strimzi/crdgenerator/CrdGenerator.java

```diff
@@ -867,6 +867,7 @@ private ObjectNode buildArraySchema(ApiVersion crApiVersion, Property property,
                 || long.class.equals(elementType)) {
             itemResult.put("type", "integer");
         } else if (Map.class.equals(elementType)) {
+            preserveUnknownFields(itemResult, elementType);
```

Comment:
At this call site we know from the line above that the element type is map, but at the call site on line 885 we don't and you've put the check on the element type within `preserveUnknownFields()`. It would probably be better to keep the check outside the method. There will be no change at this call site, but and if wrapping the call site on 885.